pkg/arvo/tests/sys/zuse/html.hoon

```diff
@@ -89,4 +90,745 @@
+      ['[\\n]' "n_string_no_quotes_with_bad_escape"]
+      ['"' "n_string_single_doublequote"]
+      ['[\'single quote\']' "n_string_single_quote"]
+      ['abc' "n_string_single_string_no_double_quotes"]
+      ['["\\' "n_string_start_escape_unclosed"]
+      ['["new' "n_string_unescaped_newline"]
+      ['line"]' "n_string_unescaped_newline"]
+      ['["\09"]' "n_string_unescaped_tab"]
+      ['"\\UA66D"' "n_string_unicode_CapitalU"]
+      ['""x' "n_string_with_trailing_garbage"]
+      ['[‚Å†]' "n_structure_U+2060_word_joined"]
+      ['Ôªø' "n_structure_UTF8_BOM_no_data"]
+      ['<.>' "n_structure_angle_bracket_."]
+      ['[<null>]' "n_structure_angle_bracket_null"]
+      ['[1]x' "n_structure_array_trailing_garbage"]
+      ['[1]]' "n_structure_array_with_extra_array_close"]
+      ['["asd]' "n_structure_array_with_unclosed_string"]
+      ['a√•' "n_structure_ascii-unicode-identifier"]
+      ['[True]' "n_structure_capitalized_True"]
+      ['1]' "n_structure_close_unopened_array"]
+      ['{"x": true,' "n_structure_comma_instead_of_closing_brace"]
+      ['[][]' "n_structure_double_array"]
+      [']' "n_structure_end_array"]
+      ['Ôª{}' "n_structure_incomplete_UTF8_BOM"]
+      ['Â' "n_structure_lone-invalid-utf-8"]
+      ['[' "n_structure_lone-open-bracket"]
+      ['["a\00a"]' "n_string_unescaped_crtl_char"]
+      ['["\\00"]' "n_string_backslash_00"]
+    ==
+:: TODO: de-json is accepting a slew of number formats it shouldn't.
+:: Tracking issue here: https://github.com/urbit/urbit/issues/1775
+:: Re-enable this test by removing the disable- prefix
+++  disable-test-reject-invalid-numbers
+    %-  zing
+    %-  turn
+    :_  |=  [exmpl=@t name=tape]
+          ^-  tang
+          =+  result=(expect-eq !>(~) !>((de-json:html exmpl)))
+          ?~  result  ~
+          `tang`[[%leaf "in {name}:"] result] 
+    :~
+      ['123\00' "n_multidigit_number_then_00"]
+      ['[1.]' "n_number_real_without_fractional_part"]
+      ['[2.e+3]' "n_number_2.e+3"]
+      ['[2.e-3]' "n_number_2.e-3"]
+      ['[2.e3]' "n_number_2.e3"]
+      ['[9.e+]' "n_number_9.e+"]
+      ['[0.3e+]' "n_number_0.3e+"]
+      ['[0.3e]' "n_number_0.3e"]
+      ['[0.e1]' "n_number_0.e1"]
+      ['[0E+]' "n_number_0_capital_E+"]
+      ['[0E]' "n_number_0_capital_E"]
+      ['[0e+]' "n_number_0e+"]
+      ['[0e]' "n_number_0e"]
+      ['[1.0e+]' "n_number_1.0e+"]
+      ['[1.0e-]' "n_number_1.0e-"]
+      ['[1.0e]' "n_number_1.0e"]
+      ['[-2.]' "n_number_-2."]
+    ==
+::  these are all inputs that should be accepted by a valid parser
+++  test-en-json-suite
+  =+  frond=frond:enjs:format
+  =+  pairs=pairs:enjs:format
+  %-  zing
+  %-  turn
+  :_  |=  [name=tape str=@t val=json]
+        ^-  tang
+        =+  result=(expect-eq !>(`val) !>((de-json:html str)))
+        ?~  result  ~
+        `tang`[[%leaf "in {name}:"] result]
+  :~
+    :*  "y_array_arraysWithSpaces"
+        '[[]   ]'
+        `json`[%a ~[[%a ~]]]
```

Comment:
Here as well, defining a mold for this `[tape cord json]` and casting to a list of that would remove the need for all this inline casting.